mockk/src/main/kotlin/io/mockk/impl/annotations/JvmMockInitializer.kt

```diff
@@ -67,9 +78,113 @@ class JvmMockInitializer(val gateway: MockKGateway) : MockKGateway.MockInitializ
         }
     }
 
+    private fun resolveInstance(target: Any, property: KProperty1<Any, Any>): Any? {
+
+        var obj: Any?
+        val objCls = property.returnType.classifier as? KClass<*>
+
+        if (objCls != null) {
+            obj = objCls.objectInstance
+
+            if (obj == null) {
+                if (objCls.constructors.isEmpty() && objCls.primaryConstructor == null) {
+                    throw MockKException("No constructors found to instantiate @InjectMockK field ${property.name}")
+                } else {
+
+                    obj = resolveByConstructorInjection(target, objCls)
+
+                    if (obj == null) {
+                        try {
+                            val noArgCtor = objCls.constructors.first {
```

Comment:
Better use firstOrNull and check from my point of view. Again lowering indentation level, as it is really high for this function